src/synthetix/StakingRewards.t.sol

```diff
@@ -270,16 +286,120 @@ contract StakingRewardsTest is Test {
 
-        vm.expectRevert("Previous rewards period must be complete before changing the duration for the new period");
-        rewards.setRewardsDuration(70 days);
+        assertEq(rewards.rewardPerTokenStored(), initialRun * (totalReward / initialDuration) * WAD / totalStake);
+        assertEq(rewards.lastUpdateTime(), block.timestamp);
+        assertEq(rewards.rewardRate(), (initialDuration - initialRun) * (totalReward / initialDuration) / newDuration);
+        assertEq(rewards.periodFinish(), block.timestamp + newDuration);
+        assertEq(rewards.rewardsDuration(), newDuration);
+
+        skip(newDuration);
+
+        rewards.exit();
+
+        uint256 rewardLeft = totalReward % initialDuration + (initialDuration - initialRun) * (totalReward / initialDuration) % newDuration; // dust lost due to rewardRate being rounded down
+        uint256 rewardPaid = totalReward - rewardLeft;
+        assertEq(rewardGem.balanceOf(address(this)), rewardPaid);
+        assertEq(rewardGem.balanceOf(address(rewards)), rewardLeft);
+        assertEq(gem.balanceOf(address(this)), totalStake);
+        assertEq(gem.balanceOf(address(rewards)), 0);
+    }
+
+    function testIncreaseRewardsDurationOnActiveDistribution() public {
+        checkChangeRewardDurationOnActiveDistribution(70 days);
+    }
+
+    function testDecreaseRewardsDurationOnActiveDistribution() public {
+        checkChangeRewardDurationOnActiveDistribution(5 days);
+    }
+
+    function testSetSameRewardsDurationOnActiveDistribution() public {
+        uint256 totalStake = 100 * WAD;
+        uint256 totalReward = 123 * WAD;
+        uint256 initialDuration = 5 days;
+        uint256 initialRun = 1 days;
+        initFarm(totalStake, totalReward, initialDuration);
+        skip(initialRun);
+
+        uint256 prevRewardRate = rewards.rewardRate();
+        uint256 prevPeriodFinish = rewards.periodFinish();
+
+        rewards.setRewardsDuration(initialDuration - initialRun); // same duration as time left
+
+        assertEq(rewards.rewardPerTokenStored(), initialRun * (totalReward / initialDuration) * WAD / totalStake);
+        assertEq(rewards.lastUpdateTime(), block.timestamp);
+        assertEq(rewards.rewardRate(), prevRewardRate);
+        assertEq(rewards.periodFinish(), prevPeriodFinish);
+        assertEq(rewards.rewardsDuration(), initialDuration - initialRun);
+    }
+
+    function testSetRewardsDurationAndNotifyRewardAmountCallOrderOnActiveDistribution() public {
+        uint256 totalStake = 100 * WAD;
+        uint256 initialReward = 123 * WAD;
+        uint256 additionalReward = 32 * WAD;
+        uint256 initialDuration = 7 days;
+        uint256 initialRun = 1 days;
+        initFarm(totalStake, initialReward, initialDuration);
+        skip(initialRun);
+
+        uint256 snapshot = vm.snapshot();
+
+        // call setRewardsDuration followed by notifyRewardAmount in the same block
+        rewards.setRewardsDuration(13 days);
```

Comment:
As we moved everything to variables, I'd say we should be doing this one as well. IMO we could have left 7 days and 1 days as they were in the tests as they are more than self explanatory. But having changed that I'd make the change here as well.

Reply:
As mentioned above, the trade-off I went for here was to use variables for numbers that are involved in some calculation (i.e. the 7-1 subtraction) but would prefer not to go overboard with using variables for everything as they indeed make the tests harder to read. Btw note that the pre-existing tests were using hardcoded numbers as well.

Reply:
Ok sounds good, resolving!